arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/ar/ArcGISArView.kt

```diff
@@ -330,58 +491,81 @@ final class ArcGISArView : FrameLayout, DefaultLifecycleObserver, Scene.OnUpdate
+            // ARCore requires camera permissions to operate. If we did not yet obtain runtime
+            // permission on Android M and above, now is a good time to ask the user for it.
+            // when the permission is requested and the user responds to the request from the OS this is executed again
+            // during onResume()
+            if (isUsingARCore && renderVideoFeed && !hasPermission(CAMERA_PERMISSION)) {
+                requestPermission(
+                    activity,
+                    CAMERA_PERMISSION,
+                    CAMERA_PERMISSION_CODE
+                )
+                return
+            }
+
+            // Request location permission if user has provided a LocationDataSource
+            locationDataSource?.let {
+                if (!hasPermission(LOCATION_PERMISSION)) {
                     requestPermission(
-                        it,
-                        CAMERA_PERMISSION,
-                        CAMERA_PERMISSION_CODE
+                        activity,
+                        LOCATION_PERMISSION,
+                        LOCATION_PERMISSION_CODE
                     )
-                    return
-                }
-
-                // when the installation is requested and the user responds to the request from the OS this is executed again
-                // during onResume()
-                if (ArCoreApk.getInstance().requestInstall(
-                        it,
-                        !arCoreInstallRequested
-                    ) == ArCoreApk.InstallStatus.INSTALL_REQUESTED
-                ) {
-                    arCoreInstallRequested = true
-                    return
+                    return@startTracking
                 }
             }
+        }
 
+        if (isUsingARCore) {
             // Create the session.
             Session(context).apply {
                 val config = Config(this)
                 config.updateMode = Config.UpdateMode.LATEST_CAMERA_IMAGE
                 config.focusMode = Config.FocusMode.AUTO
                 this.configure(config)
-                arSceneView.setupSession(this)
+                arSceneView?.setupSession(this)
             }
-
-            arSceneView.scene.addOnUpdateListener(this)
-        } catch (e: Exception) {
-            error = e
-        }
-
-        if (error != null) {
-            Log.e(logTag, error!!.message)
-            initializationStatus = ArcGISArViewState.INITIALIZATION_FAILURE
+            arSceneView?.scene?.addOnUpdateListener(this)
+            arSceneView?.resume()
+            sceneView.isManualRenderingEnabled = true
         } else {
-            arSceneView.resume()
-            sceneView.resume()
-            initializationStatus = ArcGISArViewState.INITIALIZED
+            removeView(arSceneView)
+            arSceneView = null
         }
+
+        sceneView.resume()
+        locationDataSource?.startAsync()
+        isTracking = isUsingARCore && locationDataSource != null
```

Comment:
On iOS, I'm not basing the value of `isTracking` on either ARKit or the presence of a location data source.  It's set to `true` in the call to `startTracking` if there are no errors.

Granted, if we're not using ARKit and we don't have a `locationDataSource`, then nothing is going to happen after `startTracking` is called.

What might be better is if `!isUsingARKit && !locationDataSource` generated an error (_In order to start tracking, you need to either be using ARKit or have set a location data source._) and set `isTracking` to `false.  Thoughts?

Reply:
If there is a location data source can you set `isTracking` in the completion handler for that?  If there is not a location data source, you can set it here, I think.

Reply:
Would it just be better to set the value in the `statusUpdatedListener`? Like:
```
LocationDataSource.StatusChangedListener {
            if (it.status == LocationDataSource.Status.FAILURE) {
                error = Exception(locationDataSource?.error)
                isTracking = isUsingARCore
            } else if(it.status == LocationDataSource.Status.STARTED) {
               isTracking = true
            }
        }
```

Reply:
Just to add, there's no completion listener or Future for this method on Android:
```
public final void startAsync() {
}
```

cc @gunt0001

Reply:
I would say setting it in the `statusUpdatedListener` is fine.